Fixed following scenarios not returning HTTP Error 503 while Jersey s…

kolkoo opened the pull request:
…hould not be ready anymore to service requests and may lead to exceptions:

1) While the Jersey servlet is being destroyed and a request is incoming
2) When a resource is removed, but Jersey is not yet republished, if a request comes that would get routed to this resource this will lead to exceptions instead of a 503 as expected.

Signed-off-by: IVAN GEORGIEV ILIEV ivan.iliev@musala.com

iilievdev committed:
Fixed following scenarios not returning HTTP Error 503 while Jersey should not be ready anymore to service requests and may lead to exceptions:
1) While the Jersey servlet is being destroyed and a request is incoming
2) When a resource is removed, but Jersey is not yet republished, if a request comes that would get routed to this resource this will lead to exceptions instead of a 503 as expected.
3) Changed Jersey init/reload logic to be around Jersey's WebComponent reference

Signed-off-by: IVAN GEORGIEV ILIEV <ivan.iliev@musala.com>

kolkoo commented:
Don't merge this yet - I will update this PR with a small fix of a bug it introduces :)

hstaudacher commented:
The tests are really hard to understand. Could you clean them up a little bit, e.g by splitting stuff out into methods with speakable names.

kolkoo commented:
Yes, I agree I will do so.

kolkoo commented:
Updated the PR, fixed a bug that was present due to it, modified the tests hopefully are now more readable, they are also more stabled as I fixed some problems in the concurrency logic in them.

kolkoo commented:
Just to mention this PR should resolve bugs like this one -> https://github.com/hstaudacher/osgi-jax-rs-connector/issues/135

hstaudacher commented:
thanks, will check it soon

kolkoo commented:
Well I totally forgot about this PR :D I think you did too? Any word on when you can take a look?

kaikreuzer commented:
@hstaudacher Any chance to merge this and include it in an upcoming release?
I sincerely hope you are not letting this project dying!
